File: lib-database-runtime/src/main/java/com/ustadmobile/lib/database/annotation/UmRestAccessible.java
package com.ustadmobile.lib.database.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;


/**
 * Used to indicate that the given method should be accessible over HTTP REST
 */
@Retention(RetentionPolicy.CLASS)
@Target(ElementType.METHOD)
public @interface UmRestAccessible {

    int timeout() default 5000;
}

File: lib-database-annotations/src/commonMain/kotlin/com/ustadmobile/lib/annotation/UmRestAccessible.kt
package com.ustadmobile.lib.database.annotation

/**
 * Used to indicate that the given method should be accessible over HTTP REST
 */
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class UmRestAccessible(val timeout: Int = 5000)
